test/integration/integration-test.js

```diff
@@ -0,0 +1,193 @@
+/* eslint-disable max-len  */
+/* eslint-disable max-nested-callbacks */
+/* eslint-disable max-params */
+'use strict';
+const _ = require('lodash');
+const assert = require('assert');
+
+const errors = require('../../src/api/common/errors');
+
+const wallet = require('./wallet');
+
+const settingsSpecification = require('../fixtures/settings-specification');
+const trustlineSpecification = require('../fixtures/trustline-specification');
+const payments = require('../fixtures/payments');
+
+const RippleAPI = require('../../src').RippleAPI;
+
+
+const TIMEOUT = 20000;   // how long before each test case times out
+const INTERVAL = 2000;   // how long to wait between checks for validated ledger
+
+
+function verifyTransaction(testcase, hash, type, options, done) {
+  const currentOptions = {
+    minLedgerVersion: options.minLedgerVersion,
+    maxLedgerVersion: testcase.api.getLedgerVersion()
```

Comment:
Do you mean why not just use options parameter as it is? 
Because it has maxLedgerVersion set to txData.LastLedgerSequence, which is somewhere in the future. And with that value I'm getting MissingLedgerHistoryError from getTransaction because it thinks that server does not have all needed history.
So we must fix getTransaction or just use current ledger version